### PR TITLE
Add keyboard PistachioMacroPad

### PR DESCRIPTION
**Pull request by 7-rate**
Add keyboard "PistachioMacroPad"

This is a numeric keypad with a rotary encoder that uses a stainless steel plate.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by zvecr — CHANGES_REQUESTED** [review]
Project name cannot be camel case per <https://docs.qmk.fm/#/hardware_keyboard_guidelines?id=naming-your-keyboardproject>. There will be more suggestions, but it might be worth a rename beforehand so they do not get lost.

**Comment by 7-rate** [comment]
Thanks, for the review.
I have already modified the Project name.

I also modified some parts of the keymap.

File: keyboards/pistachio_mp/config.h [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+/* nKey Rollover */
+#define FORCE_NKRO
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+
+
+#define TAPPING_TERM 170
```
Comment (zvecr) [inline keyboards/pistachio_mp/config.h:82]:
```suggestion
```

File: keyboards/pistachio_mp/config.h [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+#define MATRIX_ROW_PINS { B4, E6, D7, C6, D4 }
+#define MATRIX_COL_PINS { B3, B1, F7, F6 }
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL */
+#define DIODE_DIRECTION ROW2COL
+
+#define RGB_DI_PIN D2
+#define RGBLED_NUM 3
+#define RGBLIGHT_LAYERS
```
Comment (zvecr) [inline keyboards/pistachio_mp/config.h:57]:
```suggestion
```

File: keyboards/pistachio_mp/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    RGB_HUD, RGB_SAD, RGB_VAD,  RGB_TOG,
+    KC_NO,   KC_NO,   KC_NO,
+    KC_NO,            KC_NO,    KC_NO
+    )
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    return true;
+}
+
```
Comment (zvecr) [inline keyboards/pistachio_mp/keymaps/default/keymap.c:48]:
```suggestion
```

File: keyboards/pistachio_mp/rules.mk [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (zvecr) [inline keyboards/pistachio_mp/rules.mk:22]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by Erovia — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by fauxpark** [comment]
Thanks!

**Pull request merged**